Assets/Mirror/Editor/EnterPlayModeSettingsCheck.cs

```diff
@@ -9,35 +9,62 @@ namespace Mirror
 {
     public class EnterPlayModeSettingsCheck : MonoBehaviour
     {
-#if UNITY_2019_3_OR_NEWER
         [InitializeOnLoadMethod]
         static void OnInitializeOnLoad()
         {
-            // check immediately on load
+#if UNITY_2019_3_OR_NEWER
+            // We can't support experimental "Enter Play Mode Options" mode
+            // Check immediately on load, and before entering play mode, and warn the user
             CheckPlayModeOptions();
+#endif
 
-            // check each time we press play. OnLoad is only called once and
-            // wouldn't detect editor-time setting changes
+            // Hook this event to see if we have a good weave every time
+            // user attempts to enter play mode or tries to do a build
             EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
         }
 
         static void OnPlayModeStateChanged(PlayModeStateChange state)
         {
-            // only check when entering play mode. no need to show it again
-            // when exiting.
-            if (state == PlayModeStateChange.EnteredPlayMode)
+            // Per Unity docs, this fires "when exiting edit mode before the Editor is in play mode".
+            // This doesn't fire when closing the editor.
+            if (state == PlayModeStateChange.ExitingEditMode)
```

Comment:
why not keep EnteredPlayMode here?
what happens in 2019.3 with domain reload, is it still detected?

Reply:
EnteredPlayMode is too late.  ExitingEditMode  = "About to Enter Play Mode".  Yes the trap for CheckPlayModeOptions for 2019.3 is still there unchanged